### PR TITLE
added external rules checkString and checkChar

### PR DESCRIPTION
**Pull request by JanPurdek**
Signed-off-by: Jan Purdek <jan.purdek@pantheon.tech>

### CODE REVIEW

File: intelij-plugin/src/main/java/tech/pantheon/yanginator/plugin/external/ExternalRules.java [hunk 1]
```diff
@@ -10,17 +10,15 @@
...
-
 import com.intellij.lang.PsiBuilder;
-import com.intellij.lang.parser.GeneratedParserUtilBase.Parser;
+
 import java.util.ArrayList;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
 
+import static com.intellij.lang.parser.GeneratedParserUtilBase.*;
```
Comment (michal-cmarada) [inline intelij-plugin/src/main/java/tech/pantheon/yanginator/plugin/external/ExternalRules.java:20]: replace * import with single line imports

File: intelij-plugin/src/main/java/tech/pantheon/yanginator/plugin/external/ExternalRules.java [hunk 2]
```diff
@@ -96,4 +94,94 @@ private static boolean unanimousResults(boolean[] listOfBooleanResults) {
...
+    public static boolean checkString(PsiBuilder psiBuilder, int level, Parser parser, String... strings) {
+        if (!recursion_guard_(psiBuilder, level, "rule")) return false;
+
+        final List<String> stringList = new ArrayList<>(List.of(strings));
+        final List<Character> charList = new ArrayList<>();
+        final Set<Character> separators = new HashSet<>(Set.of('\n', '\r', ' ', '\t'));
+
+        // 0xFFFD is for characters that can't be recognized by idea and are replaced by this value
+        charList.add((char) 0xFFFD);
+        for (String s : stringList) {
```
Comment (michal-cmarada) [inline intelij-plugin/src/main/java/tech/pantheon/yanginator/plugin/external/ExternalRules.java:120]: looks like duplicate code, please extract this to private static method to remove duplications

**Review by michal-cmarada — CHANGES_REQUESTED** [review]

File: intelij-plugin/src/main/java/tech/pantheon/yanginator/plugin/external/ExternalRules.java (hunk 1 above)
Comment (JanPurdek) [reply intelij-plugin/src/main/java/tech/pantheon/yanginator/plugin/external/ExternalRules.java]: Done.

File: intelij-plugin/src/main/java/tech/pantheon/yanginator/plugin/external/ExternalRules.java (hunk 2 above)
Comment (JanPurdek) [reply intelij-plugin/src/main/java/tech/pantheon/yanginator/plugin/external/ExternalRules.java]: Done.

**Pull request merged**